src/storage-controller/src/lib.rs

```diff
@@ -341,6 +341,8 @@ where
             self.run_ingestion(id)
                 .expect("cannot fail to run ingestions now");
         }
+
+        self.storage_collections.allow_writes().await;
```

Comment:
We decided to remove the ability to come out of read-only mode within process (for now), so this code and the other `allow_writes()` code paths will go away. Sidestepping this question at least... 😅 

Still trying to answer your question: I didn't think it would matter much wether we do storage collections first or last, but after your remark I'd move it to the top. If we didn't remove this code altogether, that is.

Reply:
Thanks, makes sense!